awx/network_ui/consumers.py

```diff
@@ -0,0 +1,277 @@
+
+    def onDeviceSelected(self, message_value, topology_id, client_id):
+        'Ignore DeviceSelected messages'
+        pass
+
+    def onDeviceUnSelected(self, message_value, topology_id, client_id):
+        'Ignore DeviceSelected messages'
+        pass
+
+    def onLinkSelected(self, message_value, topology_id, client_id):
+        'Ignore LinkSelected messages'
+        pass
+
+    def onLinkUnSelected(self, message_value, topology_id, client_id):
+        'Ignore LinkSelected messages'
+        pass
+
+    def onMultipleMessage(self, message_value, topology_id, client_id):
+        for message in message_value['messages']:
+            handler = self.get_handler(message['msg_type'])
+            if handler is not None:
+                handler(message, topology_id, client_id)
+            else:
+                logger.warning("Unsupported message %s", message['msg_type'])
+
+
+persistence = _Persistence()
+
+
+# UI Channel Events
+
+@channel_session
+def ws_connect(message):
+    # Accept connection
+    data = urlparse.parse_qs(message.content['query_string'])
+    inventory_id = parse_inventory_id(data)
+    topology_ids = list(TopologyInventory.objects.filter(inventory_id=inventory_id).values_list('topology_id', flat=True))
+    topology_id = 0
+    if len(topology_ids) > 0:
+        topology_id = topology_ids[0]
+    if topology_id:
+        topology = Topology.objects.get(topology_id=topology_id)
+    else:
+        topology = Topology(name="topology", scale=1.0, panX=0, panY=0)
+        topology.save()
+        TopologyInventory(inventory_id=inventory_id, topology_id=topology.topology_id).save()
+    topology_id = topology.topology_id
+    message.channel_session['topology_id'] = topology_id
+    Group("topology-%s" % topology_id).add(message.reply_channel)
+    client = Client()
+    client.save()
+    message.channel_session['client_id'] = client.pk
+    Group("client-%s" % client.pk).add(message.reply_channel)
+    message.reply_channel.send({"text": json.dumps(["id", client.pk])})
+    message.reply_channel.send({"text": json.dumps(["topology_id", topology_id])})
+    topology_data = transform_dict(dict(topology_id='topology_id',
+                                        name='name',
+                                        panX='panX',
+                                        panY='panY',
+                                        scale='scale',
+                                        link_id_seq='link_id_seq',
+                                        device_id_seq='device_id_seq'), topology.__dict__)
+
+    message.reply_channel.send({"text": json.dumps(["Topology", topology_data])})
+    send_snapshot(message.reply_channel, topology_id)
+
+
+def send_snapshot(channel, topology_id):
+    interfaces = defaultdict(list)
+
+    for i in (Interface.objects
+              .filter(device__topology_id=topology_id)
+              .values()):
+        interfaces[i['device_id']].append(i)
+    devices = list(Device.objects
```

Comment:
This would be easier to scan/read as 1 line

Reply:
I agree.